identity/did/service.go

```diff
@@ -41,34 +51,104 @@ func CalculateCreatedAddress(address common.Address, nonce uint64) common.Addres
+			return
+		}
+
+		log.Infof("Sent off identity creation Ethereum transaction hash [%x] and Nonce [%v] and Check [%v]", ethTX.Hash(), ethTX.Nonce(), ethTX.CheckNonce())
+		log.Infof("Transfer pending: 0x%x\n", ethTX.Hash())
+
+		res, err := ethereum.QueueEthTXStatusTask(accountID, txID, ethTX.Hash(), s.queue)
+		if err != nil {
+			errOut <- err
+			return
+		}
+
+		_, err = res.Get(txMan.GetDefaultTaskTimeout())
+		if err != nil {
+			errOut <- err
+			return
+		}
+		errOut <- nil
+	}
+
+}
+
+func (s *service) calculateIdentityAddress(ctx context.Context) (*common.Address, error) {
+	factoryAddress := getFactoryAddress()
+	nonce, err := s.getNonceAt(ctx, factoryAddress)
 	if err != nil {
-		log.Infof("Failed to get txOpts from Ethereum client: %v", err)
-		return nil, nil, err
+		return nil, err
 	}
 
-	tx, err := s.client.SubmitTransactionWithRetries(s.factoryContract.CreateIdentity, opts)
+	identityAddress := CalculateCreatedAddress(factoryAddress, nonce)
+	log.Infof("Calculated Address of the identity contract: 0x%x\n", identityAddress)
+	return &identityAddress, nil
+}
+
+func (s *service) isIdentityContract(identityAddress common.Address) error {
+	contractCode, err := s.client.GetEthClient().CodeAt(context.Background(), identityAddress, nil)
 	if err != nil {
-		log.Infof("Failed to send identity for creation [txHash: %s] : %v", tx.Hash(), err)
-		return nil, nil, err
+		return err
 	}
 
-	log.Infof("Sent off identity creation Ethereum transaction hash [%x] and Nonce [%v] and Check [%v]", tx.Hash(), tx.Nonce(), tx.CheckNonce())
-	log.Infof("Transfer pending: 0x%x\n", tx.Hash())
+	deployedContractByte := common.Bytes2Hex(contractCode)
+	identityContractByte := getIdentityByteCode()[2:] // remove 0x prefix
+	if deployedContractByte == identityContractByte {
+		return nil
+	}
 
-	// TODO use transactionStatusTask and following code as a statusHandler
+	return errors.New("deployed identity contract bytecode not correct")
 
-	factoryAddress := getFactoryAddress()
-	nonce, err := s.getNonceAt(ctx, factoryAddress)
+}
+
+func (s *service) CreateIdentity(ctx context.Context) (did *DID, err error) {
+	tc, err := contextutil.Account(ctx)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
-	identityAddress := CalculateCreatedAddress(factoryAddress, nonce)
-	log.Infof("Address of created identity contract: 0x%x\n", identityAddress)
+	opts, err := s.client.GetTxOpts(tc.GetEthereumDefaultAccountName())
+	if err != nil {
+		log.Infof("Failed to get txOpts from Ethereum client: %v", err)
+		return nil, err
+	}
+
+	idConfig, err := contextutil.Self(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	identityAddress, err := s.calculateIdentityAddress(ctx)
+	if err != nil {
+		return nil, err
+	}
 
-	did := NewDID(identityAddress)
+	txID, done, err := s.txManager.ExecuteWithinTX(context.Background(), idConfig.ID, uuid.Nil, "Check TX status", s.createIdentityTX(opts))
```

Comment:
```suggestion
	txID, done, err := s.txManager.ExecuteWithinTX(context.Background(), idConfig.ID, uuid.Nil, "Identity Creation TX status", s.createIdentityTX(opts))
```
Or similar, so there is context of which action has invoked the execute. Ideally we would use constants here. What do you think?

Reply:
I will add constant. Good idea.